numpy/core/tests/test_ufunc.py

```diff
@@ -612,6 +612,13 @@ def test_true_divide(self):
         assert_(res == 0.0)
         assert_(res.dtype.name == 'float64')
 
+    def test_trunc_expected_dtypes(self):
+        # Supported dtypes by trunc.
+        for tc in 'bBhHiIlLqQefdgO':
```

Comment:
What about boolean `?`?  (What should that actually do?)

Other than that, the only question is whether we should not aim for fixing this for the whole class of truncating functions?

Reply:
Ah. Right now, `bool` get promoted which we don't want. Will add a loop entry for `bool`.

As for others, is it ok to do a follow-up PR.

Apologies for the delayed response. (Will be more active henceforth :))

Thanks!